src/test/java/ca/yorku/cse2311/tab2pdf/parser/DoubleBarParserTest.java

```diff
@@ -0,0 +1,110 @@
+package ca.yorku.cse2311.tab2pdf.parser;
+
+import ca.yorku.cse2311.tab2pdf.model.DoubleBar;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class DoubleBarParserTest {
+
+    /**
+     * None of these lines should parse into a Space
+     */
+    public static final String[] INVALID_LINES = {
+            "-----1-----1-----1-----1-|-----1-----1-----1-----1-|"
+            , "1-----1-----1-----1-|-----1-----1-----1-----1-|"
+            , "-|----1-----1-----1-|-----1-----1-----1-----1-|"
+            , "**||-----1-----1-----1-|-----1-----1-----1-----1-|"
+            , "|||3-----1-----1-----1-|-----1-----1-----1-----1-|"
+            , "|-----1-----1-----1-|-----1-----1-----1-----1-|"
+    };
+
+    /**
+     * All of these lines should parse into a space
+     */
+    public static final String[] VALID_LINES = {
+            "||*-----1-----1-----1-|-----1-----1-----1-----1-|"
+            , "*||----1-----1-----1-|-----1-----1-----1-----1-|"
+            , "*|5*---1-----1-----1-|-----1-----1-----1-----1-|"
+            , "*||*--1-----1-----1-|-----1-----1-----1-----1-|"
+            , "|2-1-----1-----1-|-----1-----1-----1-----1-|"
+            , "||-----------------------------"
+    };
+
+    public static final DoubleBar[] VALID_RESPONSES = {
+            new DoubleBar(1, true, false),
+            new DoubleBar(1, false, true),
+            new DoubleBar(5, true, true),
+            new DoubleBar(1, true, true),
+            new DoubleBar(2, false, false),
+            new DoubleBar()
+    };
+
+    private DoubleBarParser parser;
+
+    @Before
+    public void setUp() {
+        parser = new DoubleBarParser();
+    }
+
+    @Test
+    public void testParseInvalidLines() {
+
+        for (String line : INVALID_LINES) {
+            try { // Go through each invalid line, hoping to throw an Exception, since they cannot be parsed
+
+                try {
+                parser.parse(line);
+                } catch (NumberFormatException e) {
+                    System.out.println(line);
+                    e.printStackTrace();
+                }
+                fail(); // Fail if we get here, since the lines are invalid, and should produce an Exception
+
+            } catch (ParseException e) {
+                /* This Exception should be triggered since we gave the parser invalid lines */
+            }
+        }
+    }
+
+    @Test
+    public void testParseValidLines() throws ParseException {
+
+        for (int i = 0; i < VALID_LINES.length; ++i) {  // Go through each valid line,
+
+            String line = VALID_LINES[i];          // grab the current line
+            try {
+                DoubleBar doubleBar = parser.parse(line);      // parse it
+                assertEquals(VALID_RESPONSES[i], doubleBar); // ensure it equals the Title in the corresponding CORRECT_TITLES index
+            } catch (NumberFormatException e) {
+                System.out.println(line);
```

Comment:
This hides NumberFormatException. Since these lines are supposed to be valid, it should be considered a fail if this exception is thrown